In [ ]:
################################### DOXX2TXT
################################### DOCX

In [7]:
import get_file_word as gf
#fileDir = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
fileDir= r'/Users/walnerpessoa/repositories/Fluxo_Editorial/words'
fileExt = r".docx"
# dt_doc,caracteres,image,table_chunks

print(gf.get_info(fileDir,fileExt,0))
print(gf.get_info(fileDir,fileExt,1))
print(gf.get_info(fileDir,fileExt,2))


#gf.get_info(fileDir,fileExt)


True
(datetime.datetime(2019, 12, 6, 13, 9), 59270, 11, 3)
True
(datetime.datetime(2020, 2, 5, 14, 54), 99723, 9, 19)
True
(datetime.datetime(2020, 2, 5, 16, 41), 73923, 23, 18)


In [8]:
# Importar bibliotecas
from os import listdir
from os.path import isfile, join
import docx2txt
import docx
import os
import glob

from os import listdir
from os.path import isfile, join

In [11]:
import glob
#targetPattern = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/*.docx"
targetPattern = r"/Users/walnerpessoa/repositorios/Fluxo_Editorial/words/*.docx"

#mypath='/Users/walnerpessoa/repositories/Fluxo_Editorial/words'
print(targetPattern)
files_array = glob.glob(targetPattern)
print(files_array)

/Users/walnerpessoa/repositorios/Fluxo_Editorial/words/*.docx
[]


In [12]:
#fileDir = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
fileDir= r'/Users/walnerpessoa/repositories/Fluxo_Editorial/words'
fileExt = r".docx"
[_ for _ in os.listdir(fileDir) if _.endswith(fileExt)]

['ID233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx',
 'ID233885-1- Metodol Rec Did - rev.docx',
 'ID233886-2- Metodol LD - rev.docx']

In [21]:
# definir uma variável para o local onde estão os arquivos Word
#mypath="/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
#mypath='/Users/walnerpessoa/repositories/Fluxo_Editorial/words'

# gerar vetor com ons nomes dos arquivos Word
onlyfiles = [f for f in listdir(fileDir) if isfile(join(fileDir, f))]

# verificar se existe o diretório
#os.path.exists('/Users/wpessoa/repositorios/Fluxo_Editorial/words')
os.path.exists('/Users/walnerpessoa/repositorios/Fluxo_Editorial/words') 

False

In [16]:
print(onlyfiles[0])
#docx2txt.process(onlyfiles[1])
doc = docx.Document(fileDir+'/'+onlyfiles[0])
#print(fileDir+'/'+onlyfiles[0])


ID233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx


In [20]:
indice = 0
image_array=[]
table_chunks = []

# abrir conecção com o arquivo Word 
doc = docx.Document(fileDir+'/'+onlyfiles[indice])

# data criação
dt_doc = doc.core_properties.created
# ler em cada parágrafo dentro do arquivo Word
paragra= [p.text for p in doc.paragraphs]

# extai a quantidade de caracteres
caracteres = docx2txt.process(fileDir+'/'+onlyfiles[indice])

# gerar um array com tamanho de todas as imagens
for image in doc.inline_shapes:
    image_array.append([image.width, image.height])
    # print (image.width, image.height)
    
# gerar um array com todas as tabelas
for table in doc.tables:
    table_chunks.append(table)

#Saída dos dados
print('Nome do diretório: ',fileDir)
print('Nome do arquivo: ',onlyfiles[indice])
print("Data de Criação do Arquivo: ", dt_doc,"\n")
print('Quantidade de carcteres:',len(caracteres))
print("Quantidade de parágrafos: ", len(paragra))
print("Quantidade de imagens no vetor de imagens/gráficos: ",len(image_array))
print("Quantidade de Tabelas: ",len(table_chunks))


Nome do diretório:  /Users/walnerpessoa/repositories/Fluxo_Editorial/words
Nome do arquivo:  ID233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx
Data de Criação do Arquivo:  2019-12-06 13:09:00 

Quantidade de carcteres: 59270
Quantidade de parágrafos:  356
Quantidade de imagens no vetor de imagens/gráficos:  11
Quantidade de Tabelas:  3


Nome do diretório:  /Users/wpessoa/repositorios/Fluxo_Editorial/words/
Nome do arquivo:  ID233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx
Quantidade de carcteres: 59270

Quantidade de parágrafos:  356
Quantidade de imagens no vetor de imagens:  11
Quantidade de Tabelas:  3
Data de Criação do Arquivo:  2019-12-06 13:09:00

In [ ]:
#####

In [ ]:
#teste com metodos da biblioteca Docx

print(doc.paragraphs[0].runs)
print(doc.paragraphs[0].text)
print(doc.paragraphs[0].runs[0].text)
print(doc.paragraphs[0].runs[0].style)
#print(doc.paragraphs[1].runs[0].text)
#print(doc.paragraphs[1].runs[0].style)
print(doc.core_properties.created)
print(doc.inline_shapes)
print(doc.inline_shapes[0].type)
print(doc.inline_shapes[0].height)
print(doc.inline_shapes[0].width)

#
